File: Core/src/main/java/jokatu/game/stage/GameOverStage.java
package jokatu.game.stage;

import jokatu.game.event.GameEvent;
import jokatu.game.input.Input;
import jokatu.game.input.UnacceptableInputException;
import jokatu.game.player.Player;
import jokatu.game.status.StandardTextStatus;
import ophelia.collections.BaseCollection;
import ophelia.event.observable.AbstractSynchronousObservable;
import org.jetbrains.annotations.NotNull;

import static ophelia.collections.set.EmptySet.emptySet;

/**
 * A {@link Stage} that happens after the game is over.  No more input should be accepted at this point.
 */
public class GameOverStage extends AbstractSynchronousObservable<GameEvent> implements Stage<GameEvent> {

	public GameOverStage(StandardTextStatus status) {
		status.setText("Game over.");
	}

	@NotNull
	@Override
	public BaseCollection<Class<? extends Input>> getAcceptedInputs() {
		return emptySet();
	}

	@Override
	public void accept(@NotNull Input input, @NotNull Player player) throws Exception {
		throw new UnacceptableInputException("The game is over!");
	}
}

File: Core/src/main/kotlin/jokatu/game/stage/GameOverStage.kt
package jokatu.game.stage

import jokatu.game.event.GameEvent
import jokatu.game.input.Input
import jokatu.game.input.UnacceptableInputException
import jokatu.game.player.Player
import jokatu.game.status.StandardTextStatus
import ophelia.event.observable.AbstractSynchronousObservable
import kotlin.reflect.KClass

/**
 * A [Stage] that happens after the game is over.  No more input should be accepted at this point.
 */
class GameOverStage(status: StandardTextStatus) : AbstractSynchronousObservable<GameEvent>(), Stage<GameEvent> {

	init {
		status.text = "Game over."
	}

	override val acceptedInputs: Collection<KClass<out Input>>
		get() = setOf()

	@Throws(Exception::class)
	override fun accept(input: Input, player: Player) {
		throw UnacceptableInputException("The game is over!")
	}
}
